### PR TITLE
Shell: Add the |& construct for piping stderr along with stdout

### PR DESCRIPTION
**Pull request by sin-ack**
When |& is typed, stderr will be piped to stdout before the actual
piping happens. This behaves basically like a 2>&1 | (and the
underlying implementation transforms it to that anyway).

### CODE REVIEW

File: Userland/Shell/Parser.cpp [hunk 1]
```diff
@@ -5,7 +5,9 @@
  */
 
 #include "Parser.h"
+#include "Kernel/API/POSIX/unistd.h"
```
Comment (ADKaster) [inline Userland/Shell/Parser.cpp:8]: :thonk: feels like a clangd driven include, <unistd.h> is already included

Comment (sin-ack) [reply Userland/Shell/Parser.cpp:8]: clangd moment

**Comment by sin-ack** [comment]
Drafted while I work on improving the syntax highlighting.

**Comment by sin-ack** [comment]
Now with 100% less janky highlighting!
image

**Review by alimpfard — COMMENTED** [review]
LGTM, I hate it :P

File: Userland/Libraries/LibLine/Editor.cpp [hunk 2]
```diff
@@ -463,6 +463,8 @@ Editor::CodepointRange Editor::byte_offset_range_to_code_point_offset_range(size
 
 void Editor::stylize(Span const& span, Style const& style)
 {
+    if (!span.is_empty())
+        return;
```
Comment (BertalanD) [inline Userland/Libraries/LibLine/Editor.cpp:467]: :thonk:

**Pull request merged**